scripts/build-dists-tarball.sh

```diff
@@ -0,0 +1,173 @@
+
+  if [ -f /etc/os-release ]; then
+    # freedesktop.org and systemd
+    . /etc/os-release
+    osname=$NAME
+    osversion=$VERSION_ID
+  elif type lsb_release >/dev/null 2>&1; then
+    # linuxbase.org
+    osname=$(lsb_release -si)
+    osversion=$(lsb_release -sr)
+  elif [ -f /etc/lsb-release ]; then
+    # For some versions of Debian/Ubuntu without lsb_release command
+    . /etc/lsb-release
+    osname=$DISTRIB_ID
+    osversion=$DISTRIB_RELEASE
+  elif [ -f /etc/debian_version ]; then
+    # Older Debian/Ubuntu/etc.
+    osname=Debian
+    osversion=$(cat /etc/debian_version)
+  elif [ -f /etc/SuSe-release ]; then
+    # Older SuSE/etc.
+    echo "Suse?"
+  elif [ -f /etc/redhat-release ]; then
+    # Older Red Hat, CentOS, etc.
+    echo "RedHat?"
+  else
+    # Fall back to uname, e.g. "Linux <version>", also works for BSD, etc.
+    osname=$(uname -s)
+    osversion=$(uname -r)
+  fi
+
+  case $(uname -m) in
+    x86_64)
+      osarch="x64"  # or AMD64 or Intel64 or whatever
+      ;;
+    i*86)
+      osarch="x86"  # or IA32 or Intel32 or whatever
+      ;;
+    *)
+      # leave ARCH as-is
+      ;;
+  esac
+
+elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW32_NT" ]; then
+  # Do something under 32 bits Windows NT platform
+  echo "32 bits Windows NT platform"
+elif [ "$(expr substr $(uname -s) 1 10)" == "MINGW64_NT" ]; then
+  # Do something under 64 bits Windows NT platform
+  echo "64 bits Windows NT platform"
+fi
+
+distFullName="$distName-$osname-$osversion-$osarch"
+echo $distFullName
+
+# Just a basic clean check
+if [ -n "$(git status --porcelain)" ]; then
+  echo "There are changes, please clean up before re-running $0";
+#  exit 1
+else
+  echo "No changes";
+fi
+
+git fetch --all --tags
+
+if [ "$1" == "latest-tag" ]; then
+  gitTagVersion=$(git describe --tags `git rev-list --tags --max-count=1`)
+  git checkout tags/$gitTagVersion -B $gitTagVersion-build
+else
+  gitTagVersion="${1}"
+fi
+
+echo $gitTagVersion
+
+git rev-parse --symbolic-full-name --abbrev-ref HEAD
+
+#if [ "git branch --list ${gitTagVersion-build}" ]; then
+#  git checkout tags/$gitTagVersion
+#else
+#  git checkout tags/$gitTagVersion -b $gitTagVersion-build
+#fi
+
+#git checkout tags/$gitTagVersion -b $gitTagVersion-build
+# As we will not push anything up using git, we don't need to track the branch?
+#git branch --set-upstream-to=origin/$gitTagVersion $gitTagVersion-build
+#git pull
+
+shaSumVal="256"
+#hashFile="$distFullName-$gitTagVersion.sha"
+#hashFile+="$shaSumVal"
+#hashFile+="sum"
+
+hashFile="$distFullName-$gitTagVersion.sha${shaSumVal}sum"
+archiveFile="$distFullName-$gitTagVersion.zip"
+echo $hashFile
+
+distDir=$(mktemp -d)
+if [ -d $distDir ]; then
+  echo "Temporary directory $distDir exists"
+else
+  echo "Temporary directory $distDir does not exist"
+  exit 2
+fi
+
+mkdir $distDir/dist
+
+cargo build --release
+
+COPY_FILES=(
+  "target/release/tari_base_node"
+  "config.toml"
+  "config/tari_config_sample.toml"
+  "log4rs.yml"
```

Comment:
This file doesn't exist

Reply:
```config.toml``` nor ```log4rs.yml``` exist in the repo, but had been in the original request.  
  
I don't think we need to include ```log4rs.yml``` has this is now included in ```tari_base_node``` and can be create on first-run. I think keeping this in one place to maintain will be easier in the long run.  
  
Might be an idea to do the same for ```config.toml```?

Reply:
There's a difference between the sample files (which typically list **all** possible options) and a minimal working file (which is what rincewind.toml is, for example).

Therefore Including `config/tari_config_sample.toml` is actually more for reference; it won't be used. It's the minimal working config that gets copied in the install script.

Reply:
In this case `log4rs-sample.yml`just so happens to double as a minimal config file.